### ISSUE TITLE
Conflict between Resource class attribute and new almanac interface endpoints

### ISSUE THREAD

**Issue by johnlinahan** [disqus/python-phabricator#53]
With [rPe502df509d5f](https://secure.phabricator.com/rPe502df509d5f8dd3106be50d7d83ac244ff96cb4), upstream phab implements almanac conduit endpoints: `almanc.interface.search` and `almanc.interface.edit`.

This now conflicts with the existing Resource class attribute `self.interface`. If you reproduce this against a Phab instance with the new `almanac.interface.*` conduit endpoints:
```
from phabricator import Phabricator
phab = Phabricator()
phab.almanac.interface.search
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'dict' object has no attribute 'search'
```
it throws an error because it is using the Resource class attribute `self.interface` instead of treating it like a normal method. This does not occur when the method name does not conflict with the Resource attributes:
```
phab.almanac.device.search
<phabricator.Resource object at 0x7f760699cf60>
```

The Resource class should not rely on certain keywords *not* becoming conduit methods.

**Comment by nmpowell** [comment]
@johnlinahan Hello, just wondering if, or how, you resolved this?

**Comment by johnlinahan** [comment]
@nmpowell unfortunately, I did not. My team has actually migrated away from Phabricator altogether.

**Comment by nmpowell** [comment]
Thanks for the response!

@thekashifmalik (sorry if you're not the correct person to message) can I suggest this issue be re-opened?

**Comment by johnlinahan** [comment]
Github, Buildkite and a bunch of smaller custom tools.

We have a monorepo and were continually running into scaling issues with both Diffusion and Harbormaster. The migration has alleviated a our old pain points.

**Comment by thekashifmalik** [comment]
This should be resolved in `0.8.1`
